rust-runtime/aws-smithy-runtime/src/client/orchestrator.rs

```diff
@@ -356,10 +357,18 @@ async fn try_attempt(
     let response = halt_on_err!([ctx] => {
         let request = ctx.take_request().expect("set during serialization");
         trace!(request = ?request, "transmitting request");
-        let connector = halt_on_err!([ctx] => runtime_components.http_connector().ok_or_else(||
-            OrchestratorError::other("No HTTP connector was available to send this request. \
-                Enable the `rustls` crate feature or set a connector to fix this.")
+        let http_client = halt_on_err!([ctx] => runtime_components.http_client().ok_or_else(||
+            OrchestratorError::other("No HTTP client was available to send this request. \
+                Enable the `rustls` crate feature or configure a HTTP client to fix this.")
         ));
+        let timeout_config = cfg.load::<TimeoutConfig>().expect("timeout config must be set");
+        let settings = {
+            let mut builder = HttpConnectorSettings::builder();
+            builder.set_connect_timeout(timeout_config.connect_timeout());
+            builder.set_read_timeout(timeout_config.read_timeout());
+            builder.build()
+        };
+        let connector = http_client.http_connector(&settings, runtime_components);
```

Comment:
I don't know what gave you the impression operation timeouts weren't working. We have integration tests for all the varieties of timeouts: https://github.com/awslabs/smithy-rs/blob/0af7920eba8d8d0dac79035c862e1dde1b377afc/aws/sdk/integration-tests/s3/tests/timeouts.rs

Reply:
none of those tests change the HTTP connector timeouts for an individual operation (and I don't think we possibly could have supported that because we didn't have a cache keyed on http configuration)